index.js

```diff
@@ -1,55 +1,199 @@
+    })
+
+    cartItemsToBuy(item)
+  })
+}
+
+// 2.1: add on click even to add the item and move from store data array to cart data array.
+
+// below i am trying to copy the format given to me from the car-item.html
+//
+// the cartItemsToBuy is the function that holds the items that a user would purchase.
+
+function cartItemsToBuy() {
+  cartItemList.innerText = ''
+
+  state.cart.forEach((item) => {
+    // 2.2: create a li in the cart area to push the slected items to.
+    const li = document.createElement('li')
+    cartItemList.appendChild(li)
+
+    const imageCart = document.createElement('img')
+    imageCart.src = `assets/icons/${item.id}.svg`
+    imageCart.alt = `${item.id}`
+    console.log(item) // helps me gain the visibility needed
+
+    imageCart.setAttribute('class', 'cart--item-icon')
+    li.appendChild(imageCart)
+
+    const p = document.createElement('p')
+    p.innerText = item.name
+    li.appendChild(p)
+
+    const removeButton = document.createElement('button')
+    const span = document.createElement('span')
+    const addButton = document.createElement('button')
+
+    addButton.setAttribute('class', 'quantity-btn add-btn center')
+    span.setAttribute('class', 'quantity-text center')
+    removeButton.setAttribute('class', 'quantity-btn remove-btn center')
+    removeButton.innerText = '-'
+    span.innerText = `${item.quantity}`
+    addButton.innerText = '+'
+    li.appendChild(addButton)
+    li.appendChild(span)
+    li.appendChild(removeButton)
+
+    // below i am trying to write the fucntion that will help me change the quantity of the item in the cart
+
+    //   3:If the item is already in the cart, increase the item's quantity in the cart or decrease
+
+    //   4: From the cart, a user can view and adjust the number of items in their cart
+    addButton.addEventListener('click', (event) => {
+      item.quantity += 1
+      cartItemsToBuy()
+      total()
+    })
+
+    removeButton.addEventListener('click', (event) => {
+      item.quantity -= 1
+      cartItemsToBuy()
+      // 5.1: for loop and an onclick event -> if data index = 0 then remove from cart data array
+      if (item.quantity === 0) {
+        const i = state.cart.indexOf(item)
+        state.cart.splice(i, 1)
+      }
+      cartItemsToBuy()
+      total()
+    })
+  })
+}
+
+function itemsInTheCart(item) {
+  console.log(state.cart)
+  //   5: If an item's quantity equals zero it is removed from the cart
+  if (state.cart.find((product) => product.name === item.name) === undefined) {
+    item.quantity = 1
+    state.cart.push(item)
+  } else {
+    item.quantity += 1
+  }
+  console.log(item)
+}
+
+// 6: A user can view the current total in their cart
+function total() {
+  const cartTotal = document.querySelector('.total-number')
+
+  let price = 0
+
+  state.cart.forEach((item) => {
+    price += item.price * item.quantity
+  })
+
+  cartTotal.innerText = `£${price}`
```

Comment:
Does this result in a very long number sometimes? It if does, maybe you'll need some extra magic to round it up or down.